File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.py
sumone = 0
for i in range(10):
    before_sumone = sumone # 더하기 전 값 저장
    sumone += int(input()) # 받아서 더함
    if sumone == 100:
        print(sumone)
        break
    if sumone > 100:
        if abs(100 - before_sumone) >= abs(sumone - 100):
            print(sumone)
            break
        else:
            print(before_sumone)
            break
if sumone < 100:
    print(sumone)

File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().split("\n").map(Number);

let sum = 0;
let result = 0;

let under = 0;
let over = 0;
for (let i = 0; i < input.length; i++) {
  if (sum <= 100) {
    // 100보다 작으면 저장 후 더하기
    under = sum;
    sum += input[i];
  } else if (sum >= 100) {
    // 100보다 크면 under over 비교 후 100에 더 가까우면서 큰것 result 저장
    over = sum;
    if (Math.abs(100 - under) >= Math.abs(100 - over)) {
      result = over;
    } else {
      result = under;
    }
    console.log(result);
    break;
  }
}

if (sum < 100) {
  console.log(sum);
}
